File: solutions/javascript/longest-common-prefix.js
function longestCommonPrefix(strs) {
  if (!strs.length) return "";
  let prefix = strs[0];
  for (let word of strs.slice(1)) {
    while (!word.startsWith(prefix)) {
      prefix = prefix.slice(0, -1);
      if (!prefix) return "";
    }
  }
  return prefix;
}
console.log(longestCommonPrefix(["flower", "flow", "flight"]));

File: solutions/cpp/longest-common-prefix.cpp
#include <iostream>
#include <vector>
using namespace std;

string longestCommonPrefix(vector<string>& strs) {
    if (strs.empty()) return "";
    string p = strs[0];
    for (auto& w : strs) {
        while (w.find(p) != 0) p = p.substr(0, p.size()-1);
        if (p.empty()) return "";
    }
    return p;
}

int main() {
    vector<string> v = {"flower","flow","flight"};
    cout << longestCommonPrefix(v);
}
